src/components/InstallButton.js

```diff
@@ -10,24 +10,56 @@ import {
   NR1_LOGIN_URL,
   NR1_GUIDED_INSTALL_NERDLET,
   NR1_PACK_DETAILS_NERDLET,
+  UTM_PARAMETERS,
+  NR1_SIGNUP_URL,
 } from '../data/constants';
 import { quickstart } from '../types';
 
+/**
+ * @param {Object} location
+ * @returns {String}
+ */
+const checkUtmParameters = (parameters) => {
+  if (!parameters) {
+    return false;
+  }
+
+  const hasUtmParameters = Object.entries(UTM_PARAMETERS).some(
+    ([key, value]) => {
+      return parameters.get(key) === value;
+    }
+  );
+
+  return hasUtmParameters;
+};
+
 /**
  * @param {String} id
  * @param {String} nerdletId
+ * @param {Boolean} hasGuidedInstall
+ * @param {Boolean} hasUtmParameters
+ * @param {String} parameters
  * @returns {String}
  */
-const createInstallLink = (id, nerdletId, hasGuidedInstall) => {
+const createInstallLink = (
+  id,
+  nerdletId,
+  hasGuidedInstall,
+  hasUtmParameters,
+  parameters
+) => {
   const platformUrl = hasGuidedInstall
     ? getGuidedInstallStackedNr1Url(nerdletId)
     : getPackNr1Url(id, nerdletId);
-  const url = new URL(
-    `?return_to=${encodeURIComponent(platformUrl)}`,
-    NR1_LOGIN_URL
-  );
 
-  return url.href;
+  const installUrl = new URL(hasUtmParameters ? NR1_SIGNUP_URL : NR1_LOGIN_URL);
+  parameters &&
+    parameters.forEach((value, key) => {
+      installUrl.searchParams.set(key, value);
+    });
```

Comment:
While this is a really nice pattern in JSX to quickly show/hide something, it can be confusing when used within normal JS. If we use a quick if statement here it really clears things up.
```suggestion
  if (parameters) {
    parameters.forEach((value, key) => {
      installUrl.searchParams.set(key, value);
    });
  }
```